### PR TITLE
test: added tests for category sidebar component

### PR DESCRIPTION
**Pull request by Diegoalbag**
## Description

Added unit tests for packages/theme/modules/catalog/category/components/sidebar/CategorySidebar.vue
## Related Issue

https://vsf.atlassian.net/jira/software/c/projects/M2/boards/51?modal=detail&selectedIssue=M2-441
## Screenshots (if appropriate):
Screen Shot 2022-05-19 at 9 43 07 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    const routerMock = useRouterMock();
+    const contextMock = useContextMock();
+    const categoryMock = useCategoryStoreMock();
+
+    useUiState.mockReturnValue(uiStateMock);
+    useUiHelpers.mockReturnValue(UiHelpersMock);
+    useRoute.mockReturnValue(routeMock);
+    useRouter.mockReturnValue(routerMock);
+    useContext.mockReturnValue(contextMock);
+    useCategoryStore.mockReturnValue(categoryMock);
```
Comment (bartoszherba) [inline packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js:50]: Can we extract common part for all tests and execute it in beforeEach?

File: packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import { womanCategoryTreeData, defaultCategoryTreeData } from '~/test-utils/mocks/categoryTreeDataMock';
+import useCategoryStoreMock from '~/test-utils/mocks/useCategoryStore';
+import { useMobileCategoryTree } from '../logic';
+
+jest.mock('~/composables', () => ({
+  ...jest.requireActual('~/composables'),
+  useUiHelpers: jest.fn(),
+  useUiState: jest.fn(),
+}));
+jest.mock('~/modules/catalog/category/stores/category', () => ({
```
Comment (bartoszherba) [inline packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js:21]: Please, separate mocks setup with empty lines for a readability

**Review by bartoszherba — CHANGES_REQUESTED** [review]
@Diegoalbag 
A few additional things to do
1. Please, rework all scripts from js to ts, I know it might be misleading that most tests are in js but this is in most cases a legacy code and was not updated yet. All new tests/mocks/utils/scripts-in-general are in TS.

2. Please, add additional tests: 
- if the user click on a category without children he will be redirected to that category
- if the user click on a category with children deeper categories will be revealed
- if the user click on the back arrow (not the go back button) menu will be hidden

File: packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js (hunk 1 above)
Comment (Diegoalbag) [reply packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js:50]: OK

File: packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js (hunk 2 above)
Comment (Diegoalbag) [reply packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js:21]: OK

Comment (bartoszherba) [reply packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js:21]: Ok, but you did not do that? Each jest.mock must be separated with an empty line.

File: packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js (hunk 1 above)
Comment (bartoszherba) [reply packages/theme/modules/catalog/category/components/sidebar/MobileCategorySidebar/__tests__/MobileCategorySidebar.spec.js:50]: Ok === you did or you will extract it?

**Review by bartoszherba — CHANGES_REQUESTED** [review]

**Comment by Frodigo** [comment]
@bartoszherba if you have a chance, please re-review this one, thanks!

**Review by bartoszherba — APPROVED** [review]

**Pull request merged**